src/sentry/integrations/github/integration.py

```diff
@@ -578,62 +597,117 @@ def dispatch(self, request: HttpRequest, pipeline: Pipeline) -> HttpResponseBase
                 lifecycle.record_failure(GitHubInstallationError.MISSING_TOKEN)
                 return error(
                     request,
-                    self.active_organization,
+                    self.active_user_organization,
                     error_short=GitHubInstallationError.MISSING_TOKEN,
                 )
 
             authenticated_user_info = get_user_info(payload["access_token"])
+
+            if self.active_user_organization.organization is not None and features.has(
+                "organizations:github-multi-org",
+                organization=self.active_user_organization.organization,
+                actor=request.user,
+            ):
+                owner_orgs = get_owner_github_organizations(access_token=payload["access_token"])
+
+                installation_info = get_eligible_multi_org_installations(
+                    access_token=payload["access_token"], owner_orgs=owner_orgs
+                )
+                pipeline.bind_state("existing_installation_info", installation_info)
+
             if "login" not in authenticated_user_info:
                 lifecycle.record_failure(GitHubInstallationError.MISSING_LOGIN)
                 return error(
                     request,
-                    self.active_organization,
+                    self.active_user_organization,
                     error_short=GitHubInstallationError.MISSING_LOGIN,
                 )
-
             pipeline.bind_state("github_authenticated_user", authenticated_user_info["login"])
             return pipeline.next_step()
 
 
+def get_owner_github_organizations(access_token: str) -> list[str]:
```

Comment:
filter through all the organization the user has access to, if their role is admin and the organization is active